### DISCUSSION TITLE
Identifying nodes of a specific grain from mesh file

### DISCUSSION THREAD

**Discussion by shahul31** [neperfepx/neper#504] [Q&A] [answered]
Hello,
Could you please help me with this, I am trying to get nodes of a specific grain of a polycrystal from the corresponding mesh file. 

I have generated a polycrystal with 51 no.of grains and meshed it, now I want to get the nodes of grain no 51 which is `elset 51` in mesh file, i.e. in .msh file at the end we can see 

> $Groups
> elset
> 51
> 1 1
> .
> .
> .
> 50 1
> _**51 2**_
> $EndGroups

Please correct me if I am wrong, I have identified in the **$Elements** section that it contains `<tag#>` of `elset` and `<elt_id_node#>` which are the nodes associated with that element of `elset`. 

If this is true, the nodes of grain no 51 I am seeking is it under element type `<elt_type> 11 2nd-order tetrahedral elements` ?? 
because the other `<elt_type>` contains `elsets` higher than 51!! which is quite confusing.

**Comment by rquey** [comment, marked as answer]
I've just added:
- `-statelset elts`, which gives the list of elements of an elset
- `-statelset nodenb`, which gives the number of nodes of an elset
- `-statelset nodes`, which gives the list of nodes of an elset

Needs version 4.4.2-26.

**Reply by shahul31** [reply]
That's wonderful, thank you very much.
